ERC20 support

TomAFrench opened the pull request:
Copy of #239 as I resynced my dev branch with yours which wiped out that PR

jefflau commented:
Hey @TomAFrench, I added another PR with a `getWeb3Read` function. Is it possible to replace your token resolver logic with this function and see if you still get those error messages on the wrong network?

TomAFrench commented:
@jefflau I've added `getWeb3Read` in now.

I was getting rate limited by Infura so I adjusted your code so that it checks if the user is on the correct network and if so just uses their web3. I've also added a temporary infura key for testing.

makoto commented:
LGTM! Thank you so much!

<img width="717" alt="Screenshot 2020-01-17 at 19 32 42" src="https://user-images.githubusercontent.com/2630/72640609-26deca00-3960-11ea-9ced-720bbe69aad1.png">
